src/server/pkg/hashtree/db.go

```diff
@@ -1216,6 +1145,32 @@ func (mq *mergePQ) next() ([]*MergeNode, error) {
 	return ns, nil
 }
 
+func (mq *mergePQ) peek() ([]*MergeNode, error) {
+	if len(mq.q) == 0 || mq.size == 0 || mq.q[1] == nil {
```

Comment:
I don't think we would want to return this error type here. ErrBreak type is intended for breaking out of callback based iteration and should not be used for indicating an actual error. I would probably just use a fmt.Errorf error here.

Reply:
Ya, I was previously using the ErrBreak here to break out of maybeProgressReader, but it makes more sense to check for the right size upfront there as well. Fixed.